corehq/apps/reports/standard/forms/reports.py

```diff
@@ -125,10 +126,13 @@ def _fmt_url(doc_id):
                 else:
                     view_name = 'download_form'
                 try:
+                    url = reverse(view_name, args=[self.domain, doc_id])
+                    if tab_link:
+                        url += tab_link
```

Comment:
nit: I believe it would be better to simply pass the link to this function instead of an optional tab parameter.
This function looks pretty custom to make form urls as of now.

If you are up for it, you could extract another smaller function that figures out the URL for the form and then use that URL later when creating the response list directly and append the tab for case when calling `_fmt_url` for view cases link.

Reply:
I've made `_fmt_url` a bit more generic by extracting the tab link and url retrieving logic out ([fb4275a](https://github.com/dimagi/commcare-hq/pull/34320/commits/fb4275a639f81cb149ea2db7fa48f518ae8b1074)). Let me know if you think it needs further refactoring.